### PR TITLE
clean up a lot of IDA to remove a lot of type parameters

### PR DESCRIPTION
**Pull request by oscardssmith**
This also makes the `u` and `du` fields into regular `Array` instead of a `NVector`. This yak had a lot of hair.

### CODE REVIEW

File: src/common_interface/integrator_types.jl [hunk 1]
```diff
@@ -182,25 +178,26 @@ mutable struct IDAIntegrator{uType,
...
+    tmp::Array{Float64, N}
+    uprev::Array{Float64, N}
     flag::Cint
     just_hit_tstop::Bool
     event_last_time::Int
     vector_event_last_time::Int
     callback_cache::CallbackCacheType
     last_event_error::Float64
+    u_nvec::NVector
+    du_nvec::NVector
```
Comment (ChrisRackauckas) [inline src/common_interface/integrator_types.jl:192]: Do these alias u and du?

**Comment by ChrisRackauckas** [comment]
Why is this change only to IDA? I would think that some of these things are changing a style choice that we would want to keep uniform to the other solvers?

File: src/common_interface/integrator_types.jl (hunk 1 above)
Comment (oscardssmith) [reply src/common_interface/integrator_types.jl:192]: yes

Comment (ChrisRackauckas) [reply src/common_interface/integrator_types.jl:192]: Maybe make a comment.

**Comment by ChrisRackauckas** [comment]
Yes, I think it would be good to further enforce Array like this everywhere. Also CVODE tests non-vector things in much more detail so that would be a good test that the right things are vec and reshaped. I think by the aliasing of NVector to the original array it works out, but it would be good to be very sure.

**Comment by oscardssmith** [comment]
I think this is ready, but if possible I would like to get this run on the MTK testsuite before merging (since I'm not 100% sure I haven't missed anything. This is a pretty big PR).

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
This all looks reasonable. Add the MTK downstream from here? I don't know if MTK tests call Sundials though.

**Pull request merged**